pkg/commands/mcp/actions/mcp.go

```diff
@@ -0,0 +1,302 @@
+			JSONRPC: JsonrpcVersion,
+			ID:      request.ID,
+			Error: map[string]interface{}{
+				"code":    InternalError,
+				"message": "An internal error occurred when trying to send the MCP request to Apono",
+				"data":    fmt.Sprintf("Failed to send HTTP request: %v", err),
+			},
+		}, EmptyErrorStatusCode
+	}
+	defer func(Body io.ReadCloser) {
+		closeErr := Body.Close()
+		if closeErr != nil {
+			utils.McpLogf("WARNING: Failed to close response body: %v", closeErr)
+		}
+	}(resp.Body)
+
+	utils.McpLogf("HTTP response status: %d", resp.StatusCode)
+
+	if resp.StatusCode == http.StatusForbidden {
+		return McpResponse{
+			JSONRPC: JsonrpcVersion,
+			ID:      request.ID,
+			Error: map[string]interface{}{
+				"code":    AuthError,
+				"message": "Authentication failed. Please run 'apono login' to authenticate.",
+				"data":    fmt.Sprintf("Status code %v - Invalid or expired authentication token", resp.StatusCode),
+			},
+		}, resp.StatusCode
+	}
+
+	// For notification methods, don't read response body - just log completion
+	if isNotificationsMethod(request.Method) {
+		utils.McpLogf("Notification method %s completed successfully with status %d", request.Method, resp.StatusCode)
+		return McpResponse{}, resp.StatusCode
+	}
+
+	responseBody, err := io.ReadAll(resp.Body)
+	if err != nil {
+		utils.McpLogf("ERROR: Failed to read response body: %v", err)
+		return McpResponse{
+			JSONRPC: JsonrpcVersion,
+			ID:      request.ID,
+			Error: map[string]interface{}{
+				"code":    InternalError,
+				"message": "An internal error occurred when trying to send the MCP request to Apono",
+				"data":    "Failed to read response body",
+			},
+		}, resp.StatusCode
+	}
+
+	utils.McpLogf("Response body: %s", string(responseBody))
+
+	var response McpResponse
+	if err := json.Unmarshal(responseBody, &response); err != nil {
+		utils.McpLogf("ERROR: Failed to unmarshal response: %v", err)
+		return McpResponse{
+			JSONRPC: JsonrpcVersion,
+			ID:      request.ID,
+			Error: map[string]interface{}{
+				"code":    InternalError,
+				"message": "An internal error occurred when trying to unmarshal the MCP response from the server",
+				"data":    "Failed to parse response",
+			},
+		}, resp.StatusCode
+	}
+
+	if !isNotificationsMethod(request.Method) {
+		response.ID = request.ID
+	}
+
+	return response, resp.StatusCode
+}
+
+func sendResponse(response McpResponse, statusCode int) {
+	responseJSON, err := json.Marshal(response)
+	if err != nil {
+		if statusCode == http.StatusForbidden {
+			errorResponse := McpResponse{
+				JSONRPC: JsonrpcVersion,
+				ID:      response.ID,
+				Error: map[string]interface{}{
+					"code":    AuthError,
+					"message": "Authentication failed. Please run 'apono login' to authenticate.",
+					"data":    fmt.Sprintf("Status code %v - Invalid or expired authentication token", statusCode),
+				},
+			}
+			errorJSON, _ := json.Marshal(errorResponse)
+			fmt.Println(string(errorJSON))
+			return
+		}
```

Comment:
dont you already handle the status code before? why handle it twice?
this methofg should only propagate the reponse and not have any logic
all apono side response parsing logic should be in the sendReqeust method